ipatests/test_xmlrpc/test_add_remove_cert_cmd.py

```diff
@@ -352,3 +376,88 @@ def delete_entity(cls):
             api.Command.host_del(TestCertManipCmdHost.entity_pkey)
         except errors.NotFound:
             pass
+
+
+@pytest.mark.tier1
+class TestCertManipIdOverride(XMLRPC_test):
+    entity_subject = u'testuser'
+    entity_principal = u'testuser'
+
+    cert_add_cmd = api.Command.idoverrideuser_add_cert
+    cert_del_cmd = api.Command.idoverrideuser_remove_cert
+
+    def del_cert_from_idoverride(self, username, view_name, cert):
+        result = self.cert_del_cmd(view_name,
+                                   username,
+                                   usercertificate=cert)
+        return dict(
+            usercertificate=result['result'].get('usercertificate', []),
+            value=result.get('value'),
+            summary=result.get('summary')
+        )
+
+    def add_cert_to_idoverride(self, username, view_name, cert):
+        result = self.cert_add_cmd(view_name,
+                                   username,
+                                   usercertificate=cert)
+        return dict(
+            usercertificate=result['result'].get('usercertificate', []),
+            value=result.get('value'),
+            summary=result.get('summary')
+        )
+
+    def test_00_add_idoverrideuser(self, testuser, idview):
+        testuser.create()
+        idview.create()
+        idview.idoverrideuser_add(testuser)
+
+    def test_01_add_cert_to_idoverride(self, testuser, idview, cert1):
+        assert_deepequal(
+            dict(usercertificate=(base64.b64decode(cert1),),
+                 summary=u'Added certificates to'
+                         ' idoverrideuser \"%s\"' % testuser.name,
+                 value=testuser.name,
+                 ),
+            self.add_cert_to_idoverride(testuser.name,
+                                        idview.cn,
+                                        cert1)
+        )
+
+    def test_02_add_second_cert_to_idoverride(self, testuser,
+                                              idview, cert1, cert2):
+        assert_deepequal(
+            dict(
+                usercertificate=(base64.b64decode(cert1),
+                                 base64.b64decode(cert2)),
+                summary=u'Added certificates to'
+                        ' idoverrideuser \"%s\"' % testuser.name,
+                value=testuser.name,
+            ),
+            self.add_cert_to_idoverride(testuser.name,
+                                        idview.cn,
+                                        cert2)
+        )
+
+    def test_03_add_the_same_cert_to_idoverride(self, testuser,
+                                                idview, cert1, cert2):
+        raises(errors.ExecutionError,
```

Comment:
I'm aware that the rest of the test uses this utility function, however I think using pytest.raises context manager adds readability here. That way the add/del cert methods for certificate could be moved to the tracker itself.